automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/AuthDomainMultiGroupsApiSpec.scala

```diff
@@ -0,0 +1,479 @@
+package org.broadinstitute.dsde.test.api.orch
+
+import akka.http.scaladsl.model.StatusCodes
+import org.broadinstitute.dsde.workbench.auth.AuthToken
+import org.broadinstitute.dsde.workbench.config.{Credentials, UserPool}
+import org.broadinstitute.dsde.workbench.fixture.{BillingFixtures, GroupFixtures, WorkspaceFixtures}
+import org.broadinstitute.dsde.workbench.service.BillingProject.BillingProjectRole
+import org.broadinstitute.dsde.workbench.service.{AclEntry, Orchestration, Rawls, WorkspaceAccessLevel}
+import org.scalatest.{FreeSpec, Matchers}
+import spray.json.{DefaultJsonProtocol, JsValue, JsonParser}
+
+import scala.util.Try
+
+
+class AuthDomainMultiGroupsApiSpec extends FreeSpec with Matchers with WorkspaceFixtures with BillingFixtures with GroupFixtures {
+
+  /*
+   * Unless otherwise declared, this auth token will be used for API calls.
+   * We are using a curator to prevent collisions with users in tests (who are Students and AuthDomainUsers), not
+   *  because we specifically need a curator.
+   */
+  val defaultUser: Credentials = UserPool.chooseCurator
+  val authTokenDefault: AuthToken = defaultUser.makeAuthToken()
+
+
+  "A workspace with an authorization domain" - {
+    "with multiple groups inside of it" - {
+
+      "can be created" in {
+
+        val user = UserPool.chooseAuthDomainUser
+        implicit val authToken: AuthToken = authTokenDefault
+
+        withGroup("GroupsApiSpec_groupOne", List(user.email)) { groupOne =>
+          withGroup("GroupsApiSpec_groupTwo", List(user.email)) { groupTwo =>
+
+            withCleanBillingProject(user) { billingProjectName =>
+              withWorkspace(billingProjectName, "GroupsApiSpec_workspace", Set(groupOne, groupTwo)) { workspaceName =>
+
+                // user is in members emails in two authdomain groups
+                groupNameToMembersEmails(groupOne) should contain allElementsOf List(user.email)
+                groupNameToMembersEmails(groupTwo) should contain allElementsOf List(user.email)
+
+                // user can access workspace and see two authdomain groups
+                val groups = Orchestration.workspaces.getAuthorizationDomainInWorkspace(billingProjectName, workspaceName)(user.makeAuthToken())
```

Comment:
Why is this being done in orchestration? Shouldn't it be through rawls?

Reply:
The tickets don't specify which service and orchestration api will still be in service after redirects. also I am leveraging existing test fixtures like `withWorkspace`, not reinventing the wheels.

Reply:
We're deciding which service the tests are for based on what Terra is calling - so if Terra is accessing Rawls for auth domain things, the tests should be put in rawls

Reply:
If Terra _is_ accessing auth domain related apis through rawls, you can drop the tests there and do the set up steps (like creating the workspace) through orch if that's easier.